app/account_linking.py

```diff
@@ -53,11 +55,14 @@ def set_plan_for_partner_user(partner_user: PartnerUser, plan: SLPlan):
                 partner_user_id=partner_user.id,
                 end_at=plan.expiration,
             )
+            agent.record_custom_event("PlanChange", {"plan": "premium"})
         else:
-            LOG.i(
-                f"Updating partner_subscription [user_id={partner_user.user_id}] [partner_id={partner_user.partner_id}]"
-            )
-            sub.end_at = plan.expiration
+            if sub.end_at != plan.expiration:
+                LOG.i(
+                    f"Updating partner_subscription [user_id={partner_user.user_id}] [partner_id={partner_user.partner_id}]"
+                )
+                agent.record_custom_event("PlanUpdateExpiration")
```

Comment:
```suggestion
                agent.record_custom_event("PlanChange", {"plan": "premium", "type": "extension"})
```
This way we can group them if we need